File: py-codellama13B-AWQ-all/taskid-156_int_to_mini_roman-gen16.py
def int_to_mini_roman(number: int) -> str:
    """
    Given a positive integer, obtain its roman numeral equivalent as a string,
    and return it in lowercase.
    Restrictions: 1 <= num <= 1000

    Examples:
    >>> int_to_mini_roman(19)
    'xix'
    >>> int_to_mini_roman(152)
    'clii'
    >>> int_to_mini_roman(426)
    'cdxxvi'
    """

    # Use a dictionary to map each number to its corresponding Roman numeral
    # symbol.
    ROMAN_NUMERALS = {
        1000: "m",
        900: "cm",
        500: "d",
        400: "cd",
        100: "c",
        90: "xc",
        50: "l",
        40: "xl",
        10: "x",
        9: "ix",
        5: "v",
        4: "iv",
        1: "i"
    }

    result = ""
    for value, numeral in ROMAN_NUMERALS.items():
        # Use the modulo operator to obtain the integer quotient and remainder.
        quotient, remainder = divmod(number, value)
        result += numeral * quotient
        number = remainder

    # Return the Roman numeral equivalent of the input integer.
    return result

File: java-codellama7B-FP-all/taskid-156_int_to_mini_roman-gen16.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // Given a positive integer, obtain its roman numeral equivalent as a string,
    // and return it in lowercase.
    // Restrictions: 1 <= num <= 1000
    // Examples:
    // >>> intToMiniRoman((19l))
    // ("xix")
    // >>> intToMiniRoman((152l))
    // ("clii")
    // >>> intToMiniRoman((426l))
    // ("cdxxvi")
    public static String intToMiniRoman(long number) {

        final List<String> romanNumerals = Arrays.asList(
            "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"
        );
        final List<Long> arabicNumerals = Arrays.asList(
            1000l, 900l, 500l, 400l, 100l, 90l, 50l, 40l, 10l, 9l, 5l, 4l, 1l
        );
        StringBuilder builder = new StringBuilder();
        int idx = 0;
        while (number > 0) {
            while (number >= arabicNumerals.get(idx)) {
                builder.append(romanNumerals.get(idx));
                number -= arabicNumerals.get(idx);
            }
            ++idx;
        }
        return builder.toString();
    }
    public static void main(String[] args) {
    assert(intToMiniRoman((19l)).equals(("xix")));
    assert(intToMiniRoman((152l)).equals(("clii")));
    assert(intToMiniRoman((251l)).equals(("ccli")));
    assert(intToMiniRoman((426l)).equals(("cdxxvi")));
    assert(intToMiniRoman((500l)).equals(("d")));
    assert(intToMiniRoman((1l)).equals(("i")));
    assert(intToMiniRoman((4l)).equals(("iv")));
    assert(intToMiniRoman((43l)).equals(("xliii")));
    assert(intToMiniRoman((90l)).equals(("xc")));
    assert(intToMiniRoman((94l)).equals(("xciv")));
    assert(intToMiniRoman((532l)).equals(("dxxxii")));
    assert(intToMiniRoman((900l)).equals(("cm")));
    assert(intToMiniRoman((994l)).equals(("cmxciv")));
    assert(intToMiniRoman((1000l)).equals(("m")));
    }

}
